cmd/ef_tests/test_runner.rs

```diff
@@ -5,27 +5,63 @@ use ethereum_rust_core::{
     rlp::decode::RLPDecode,
     rlp::encode::RLPEncode,
     types::{Account as CoreAccount, Block as CoreBlock, Transaction as CoreTransaction},
-    Address,
+    Address, U256,
 };
 use ethereum_rust_evm::{
-    apply_state_transitions, evm_state, execute_tx, process_withdrawals, EvmState, SpecId,
+    apply_state_transitions, beacon_root_contract_call, evm_state, execute_block, execute_tx,
+    process_withdrawals, EvmState, SpecId,
 };
 use ethereum_rust_storage::{EngineType, Store};
 
+/// Tests the execute_block function
+pub fn execute_test2(test: &TestUnit) {
+    // Build pre state
+    let mut evm_state = build_evm_state_from_prestate(&test.pre);
+    apply_state_transitions(&mut evm_state).expect("Failed to update DB state");
+    let blocks = test.blocks.clone();
+    // Execute all blocks in test
+    for block in blocks.iter() {
+        execute_block(
+            &block.block().clone().into(),
+            &mut evm_state,
+            SpecId::CANCUN,
+        )
+        .expect("Error executing block");
```

Comment:
We should check if the test expected an exception here and handle it like we do in execute test
```suggestion
       let execution_result = execute_block(
            &block.block().clone().into(),
            &mut evm_state,
            SpecId::CANCUN,
        );
        
        if block.expect_exception.is_some() {
                assert!(
                    execution_result.is_err(),
                    "Expected transaction execution to fail on test: {}",
                    test_key
                )
            } else {
                assert!(
                    execution_result.is_ok(),
                    "Transaction execution failed on test: {} with error: {}",
                    test_key,
                    execution_result.unwrap_err()
                )
            }
```